freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UmsUsersStateProvider.java

```diff
@@ -44,16 +41,8 @@
     private GrpcUmsClient grpcUmsClient;
 
     public Map<String, UsersState> getEnvToUmsUsersStateMap(String accountId, String actorCrn, Set<String> environmentCrns,
-                                                            Set<String> userCrns, Set<String> machineUserCrns) {
+                                                            Set<String> userCrns, Set<String> machineUserCrns, Optional<String> requestIdOptional) {
```

Comment:
So again, I don't understand the calling convention, but it seems to me that passing the request id everywhere is prone to create problems. What you want is that every grpc call to have a request id. It is better if you can ensure this in the request builder. The request builder can look at a thread local to see if the thread is part of an ongoing request. If not, generate a new request ID. I'm not sure if this is tracked but it is definitely worth a jira if there is no jira for that.

Reply:
I agree. requestId handling could be pushed to the GrpcUmsClient class. I opened CB-4021